// pots/dist/controller/view.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.view = void 0;
const cli_table3_1 = __importDefault(require("cli-table3"));
const database_1 = __importDefault(require("../config/database"));
function view(flags) {
    const pots = database_1.default.Pot.findAll({ keepOpen: true })
        .map(pot => pot.name);
    const transactions = database_1.default.Transaction.findAll({ keepOpen: true })
        .filter((trans) => trans.pot);
    const sumsByPot = transactions.reduce((acc, trans) => {
        const current = acc[trans.pot.name] || 0;
        acc[trans.pot.name] = current + trans.amount;
        return acc;
    }, {});
    const columns = ["Transactions", ...pots];
    const sums = ["Totals", ...pots.map(pot => (sumsByPot[pot] || 0).toFixed(2))];
    const table = new cli_table3_1.default({
        colWidths: [35],
        chars: {
            "mid": " ",
            "right-mid": "│",
            "mid-mid": "│",
        },
        colAligns: ["left", ...new Array(columns.length - 1).fill("right")],
        style: { head: [] },
        wordWrap: true,
    });
    table.push(columns);
    table.push(sums);
    transactions.forEach((trans) => {
        const potIndex = columns.indexOf(trans.pot.name);
        const arr = new Array(columns.length).fill("");
        arr[0] = {
            content: trans.name,
            chars: { "mid": "─", "left-mid": "├" },
        };
        arr[potIndex] = trans.amount.toFixed(2);
        arr[1] = {
            content: arr[1],
            chars: { "mid-mid": "┤" },
        };
        table.push(arr);
    });
    database_1.default.close();
    return table.toString();
}
exports.view = view;
