// pots/dist/commands/pot/add.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const core_1 = require("@oclif/core");
const pot_1 = require("../../controller/pot");
class Add extends core_1.Command {
    static summary = "Add a pot.";
    static examples = [`$ <%= config.bin %> <%= command.id %>
  $ <%= config.bin %> <%= command.id %> -n life
  $ <%= config.bin %> <%= command.id %> -n life --color #ff0000 --color-bg dddddd --dryrun
  `];
    static flags = {
        name: core_1.Flags.string({
            char: "n",
            description: "Name of the pot",
            aliases: ["p", "pot", "text"],
        }),
    };
    async run() {
        const { flags } = await this.parse(Add);
        await (0, pot_1.add)(flags).catch(err => {
            if (err.name === "SequelizeUniqueConstraintError")
                this.error(`Name already exists in the system.`);
            else
                this.error(`${err.name}: ${err.message}`);
        }).then(name => this.log(`Successfully created pot '${name}'.`));
    }
}
exports.default = Add;
