// pots/dist/commands/pot/edit.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const core_1 = require("@oclif/core");
const pot_1 = require("../../controller/pot");
class Edit extends core_1.Command {
    static summary = "Update a pot.";
    static examples = [`$ <%= config.bin %> <%= command.id %>
  $ <%= config.bin %> <%= command.id %> -n life
  $ <%= config.bin %> <%= command.id %> -n life -N Leben
  `];
    static flags = {
        name: core_1.Flags.string({
            char: "n",
            description: "Name of the pot",
            aliases: ["p", "pot", "text"],
        }),
        newName: core_1.Flags.string({
            char: "N",
            description: "New name of the pot",
        }),
    };
    async run() {
        const { flags } = await this.parse(Edit);
        await (0, pot_1.edit)(flags).catch(err => {
            this.error(`${err.message}`);
        }).then(properties => {
            this.log(`Successfully updated pots properties: ${properties.map(p => `'${p}'`)}.`);
        });
    }
}
exports.default = Edit;
