// pots/dist/commands/trans/add.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const core_1 = require("@oclif/core");
const transaction_1 = require("../../controller/transaction");
class Add extends core_1.Command {
    static summary = "Add a transaction.";
    static examples = [`$ <%= config.bin %> <%= command.id %>
$ <%= config.bin %> <%= command.id %> -n "Edeka" -p life -a -22.67
$ <%= config.bin %> <%= command.id %> -n "Edeka" -p life -a -22.67 -t groceries -d yesterday --color #ff0000 --color-bg dddddd --dryrun
`];
    static flags = {
        name: core_1.Flags.string({
            char: "n",
            description: "Name of the transaction",
            aliases: ["text"],
        }),
        pot: core_1.Flags.string({
            char: "p",
            description: "Pot that the transaction belongs to",
        }),
        date: core_1.Flags.string({
            char: "d",
            description: "[default: today] Date of the transaction",
        }),
        amount: core_1.Flags.string({
            char: "a",
            description: "Transaction size, specify negative with minus (-)",
            aliases: ["size"],
        }),
    };
    async run() {
        const { flags } = await this.parse(Add);
        await (0, transaction_1.add)(flags)
            .catch(err => this.error(err.message));
    }
}
exports.default = Add;
