// pots/dist/commands/trans/edit.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const core_1 = require("@oclif/core");
const transaction_1 = require("../../controller/transaction");
class Edit extends core_1.Command {
    static summary = "edit a transaction.";
    static examples = [`$ <%= config.bin %> <%= command.id %> -N "Edeka"
$ <%= config.bin %> <%= command.id %> -N "Edeka" -P life -A -22.67 -D yesterday --color #ff0000 --color-bg dddddd --dryrun
`];
    static flags = {
        name: core_1.Flags.string({
            char: "N",
            description: "New name of the transaction",
        }),
        pot: core_1.Flags.string({
            char: "P",
            description: "New pot that the transaction will belong to",
        }),
        date: core_1.Flags.string({
            char: "D",
            description: "[default: today] New date of the transaction",
        }),
        amount: core_1.Flags.string({
            char: "A",
            description: "New transaction amount, specify negative with minus (-)",
        }),
    };
    async run() {
        const { flags } = await this.parse(Edit);
        await (0, transaction_1.edit)(flags)
            .catch(err => this.error(err.message))
            .then(properties => {
            this.log(`Successfully updated transactions properties: ${properties.map(p => `'${p}'`)}.`);
        });
    }
}
exports.default = Edit;
